### ISSUE TITLE
S-GA msColl <delSpan> to <anchor> trouble

### ISSUE THREAD

**Issue by ebeshero** [FrankensteinVariorum/collationWorkspace#11]
## Missing deletion info in msColl (Shelley-Godwin Archive manuscript notebook)
Discovered while working on C20.
We are losing some collation information about deleted passages by having the Python script skip over `<delSpan/>` and `<anchor/>` elements. In the msColl, `<delSpan>` contains a `@spanTo` attribute that points to an anchor a little later in the text, and this is used to mark long deleted passages around lines. `<delSpan/>` and `<anchor/>` are self-closing in the original. In our flattened version, optimized for collation, a sample delSpan looks like this. 

```xml
`<delSpan rend="strikethrough" spanTo="#c57-0031.06"/>&amp;  
<lb n="c57-0031__main__11"/>
   untill very lately had lived 
<anchor xml:id="c57-0031.06"/> 
```

The collation output lacks *any* information that this is a deleted passage because both `<delSpan>` and `<anchor>` elements are in the ignore list in the Python script. 

```xml
<rdgGrp n="['france', 'and', 'untill', 'very', 'lately', 'had', 'lived', 'where for many years he had lived']">
	<rdg wit="fMS">France &amp;&lt;lb n="c57-0031__main__11"/&gt; untill very lately had lived &lt;longToken&gt;where for many years &lt;lb n="c57-0031__main__12"/&gt;he had lived&lt;/longToken&gt;</rdg>
</rdgGrp>
```

We are currently handling deletions as *whole elements* `<del>....</del>` that we treat as automatic longTokens complete/irreducible in themselves. That's not the case with `<delSpan/>` to `<anchor/>` passages. But these, too, should be treated semantically as deletions. 

## What to do about it?

We should not ignore these `<delSpan/>` to `<anchor/>` passages. We have to do something.
1. Minimal: Remove `<delSpan/>` and `<anchor/>` from the Python ignore list. `<anchor>` is used very frequently in msColl, and including it will likely be helpful as signal markers of locations in the edition. But that doesn't address missing the semantic info of normalizing delSpans as deletions.

2. Optimal: 
     * In pre-processing the msColl, **raise** `<delSpan/>` to `<anchor/>` in an obviously altered form. 
          * maybe like this:
           `<sga-delSpan rend="strikeThrough" spanTo="#c57-0031.06">....</sga-delSpan>`
          * or **wrap** something around the passage `<sga-delSpan>.....</sga-delSpan>` to contain the effected area and treat as a longToken.  
     * In Python: 
         * Remove `<delSpan>` from Python script.
         * Remove `<anchor>` from the ignore list
         * Make sure `<sga-delSpan>` is included in the `inlineVariationEvent` list
     * 

### Revising collation data
This will require going back and re-running the collation on past units. Hopefully, disturbance to alignment is minimal, and we see some improvements if we're effectively laying down some collation longToken units. 

## Revising the XSLT post-production pipeline
We need to remove any extra "longToken scaffolding" we set in place. A good consequence is preserving more of the SGA elements in the msColl output, to help with xpointers.

@Yuying-Jin @raffazizzi @arrowarchive @nhammer514

**Comment by ebeshero** [comment]
The problem with this is that several times in the SGA msColl, we see delSpans that effectively wrap inner delSpans, at least at one point in c56, we can find delSpans 3 layers "deep": a long passage struck out with a vertical line, inside which are one or two more struck-through passages. With some clever XPath and XSLT, we may be able to isolate those for special handling, but we still need to deal with them in post-production. And the consequences of creating very long tokens out of very long *ambiguously* deleted passages (the ones with simple vertical lines) might not be good for the collation alignment.

So, due to the complex nature of hand-marks over the S-GA, maybe we should err on the side of caution with `<delSpan>` elements. We really must output them as part of the edition data, but maybe we should not automate handling these as del-style longTokens. We can, as needed, wrap `<longToken>` elements around such passages (as we have been doing) to improve alignments.  We would normalize them as `''` (empty tokens), just like `<lb/>`. 

@Yuying-Jin @raffazizzi @arrowarchive @nhammer514

**Comment by ebeshero** [comment]
@Yuying-Jin @raffazizzi @arrowarchive @nhammer514 : I'm thinking in this issue about a big problem I recognized this evening. I think I'm going to take a minimal approach with it, and just make sure we are outputting the `<delSpan>` and `<anchor>` elements. They are currently in the "ignore" list in our Python script, which means they never come out in the collated edition files. That's maybe a problem, because we lose evidence in the Variorum that probably hundreds of passages were marked for deletion. 

Yuying and Raff, I think if we simply make sure that `<delSpan> ` and `<anchor>` elements are output in our edition data, this will make it easier to handle pointing to the S-GA edition: we'll have more locational markers to help. And we could do something in postProcessing (watching for "delSpan to anchor" passages) to make sure we know these are deleted.

**Comment by ebeshero** [comment]
@nhammer514 @arrowarchive If we have to redo some collation units, we'll definitely preserve your `<longToken>` elements that you've already set in place! GitHub makes this easy, and I've already updated Rachel's C-20 in progress with the revised Python collation output. 

I did some work in a branch to see how dramatically anything changes in the collation processing from outputting the `<delSpan>` and `<anchor>` elements. And the changes are pretty minimal so far.
